graph references in Cypher 25 (#1295)
neo4j/docs-cypher: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc

label:updated[]
[source, cypher, role="noheader"]
----
USE graph.byName('composite.with.dot.constituent')
USE graph.propertiesByName('composite.with.dot.constituent')
----
|

> In Cypher 25, xref::values-and-types/graph-references.adoc[Graph references] in arguments of the functions xref:functions/graph.adoc#functions-graph-byname[`graph.byName`] and xref:functions/graph.adoc#functions-graph-propertiesByName[`graph.propertiesByName`] now require no syntactic quotes.

Reviewer: ```suggestion
xref::values-and-types/graph-references.adoc[Graph references] in the arguments of the functions xref:functions/graph.adoc#functions-graph-byname[`graph.byName`] and xref:functions/graph.adoc#functions-graph-propertiesByName[`graph.propertiesByName`] now require no syntactic quotes.
```

> In Cypher 5, there was some special logic for composite databases or constituents with dots, which required quotes around those name parts in order to find the name, e.g. `USE graph.byName('`composite.with.dot++`.constituent')++`.

Reviewer: ```suggestion
In Cypher 5, if a composite database or constituent name contains dots, those name parts have to be wrapped in quotes to resolve the name correctly,  e.g., `USE graph.byName('`composite.with.dot++`.constituent')++`.
```

> label:updated[]

Reviewer 2: since this is talking about deprecations/removals it should probably not be under the `updated in Cypher 25` section, the deprecated bit should probably be in a `Deprecated features` section (not related to Cypher 25) and then in `Removed in Cypher 25` for the breaking change part 🤔

[source, cypher, role="noheader"]
----
USE graph.propertiesByName('`composite.with.dot`.constituent')
----
|

> The graph name ``composite.with.dot`.constituent` is deprecated. It is replaced by ``composite.with.dot`.constituent`.

Reviewer 2: isn't your replacement the exact same thing as the deprecated one? 🤔

Reviewer 2: ```suggestion
The graph name ``composite.with.dot`.constituent` is deprecated. It is replaced by ``composite.with.dot.constituent``.
```

Reviewer 2: I don't think just the fix above of moving a ` will help but there is also in general something up with the formatting:
![image](https://github.com/user-attachments/assets/6c83663e-1479-4dd8-829e-d95e89a9bdc6)

In Cypher 5, there was some special logic for composite databases or constituents with dots,
which required quotes around those name parts in order to find the name. This deprecations apply to xref::values-and-types/graph-references.adoc[Graph references] in arguments of the functions xref:functions/graph.adoc#functions-graph-byname[`graph.byName`] and xref:functions/graph.adoc#functions-graph-propertiesByName[`graph.propertiesByName`]. In Cypher 25 the graph references can be passed in without quoting

> as in the name column from `SHOW ALIASES FOR DATABASE`.Note that escaping graph names within the graph functions string argument is not supported in Cypher 5.

Reviewer 2: ```suggestion
In Cypher 5, there was some special logic for composite databases or constituents with dots, which required quotes around those name parts in order to find the name.
This deprecations apply to xref::values-and-types/graph-references.adoc[Graph references] in arguments of the functions xref:functions/graph.adoc#functions-graph-byname[`graph.byName`] and xref:functions/graph.adoc#functions-graph-propertiesByName[`graph.propertiesByName`].
In Cypher 25 the graph references can be passed in without quoting as in the name column from `SHOW ALIASES FOR DATABASE`.
Note that escaping graph names within the graph functions string argument is not supported in Cypher 5.
```
Usually we split the lines between sentences and not in the middle of sentences

Author: Ah...I guess there are no deprecations in cypher 25 yet. So this is about what we newly support. I'll try to reformulate this a little.

Reviewer 2: I guess we would want to also add something for the removal of the old deprecation and introduction of the new + the breaking change, might be easier to just have that in this PR as well to not need 2 graph ref related cypher manual docs PRs 🤔

Author: Woudn't it be better to port Louise's changes to this branch instead of doing them twice?

Author: We discussed that we can take over the deprecations from https://github.com/neo4j/docs-status-codes/pull/340/ after that one is merged. This can also be in a follow up PR.
